### PR TITLE
Adding merge_deps system in the template.

### PR DESCRIPTION
**Pull request by nicolasnoble**

### CODE REVIEW

**Comment by dgquintas** [comment]
Lots of tests failures :(

**Comment by dgquintas** [comment]
LGTM.

offtopic: we seem to be flaky again :(

**Comment by vjpai** [comment]
Undo LGTM.  Change on build.yaml needs explanation or fixing.

File: build.yaml [hunk 1]
```diff
@@ -810,13 +817,14 @@ libs:
   src:
   - src/cpp/common/insecure_create_auth_context.cc
   deps:
-  - gpr
-  - grpc_unsecure
-  baselib: true
+  - grpc
```
Comment (vjpai) [inline build.yaml]: Why did this change from grpc_unsecure to grpc ?

Comment (nicolasnoble) [inline build.yaml]: That's most definitely a typo.

**Comment by jtattermusch** [comment]
Please add a new LGTM once fixed.

**Comment by dgquintas** [comment]
is this PR still relevant. Needs a merge.

**Comment by dgquintas** [comment]
ping.

**Pull request closed without merging**